GVRf/Framework/src/org/gearvrf/GVRViewManager.java

```diff
@@ -316,28 +338,131 @@ public void run() {
                 final Bitmap capturedBitmap = generateBitmap(byteArray, width,
                         height);
                 callback.onScreenCaptured(capturedBitmap);
+                capturedBitmap.recycle();
```

Comment:
Oh, please don't do this! It's harmless (though not necessary) if (and only if) the app does not use the `Bitmap` after the callback returns - but you have no way of knowing that this is the case! The app may do _anything_ besides save-to-disk-during-the-callback - imagine the app uploads the screenshot to a queue of images to be sent to a server somewhere,which is being consumed by Yet Another thread.